### PR TITLE
Allows AIs to see visible actions through cameras

### PR DESCRIPTION
**Pull request by JustSumBody**
Now with 50% less valid hunting.
Basically I moved the eye's hearer to be default but without the ability to read lips unless you choose the malf module/var edit 
The biggest concern I could see with this is stuff that uses visible_message when it shouldn't.

### CODE REVIEW

**Comment by Ephx1** [comment]
Things to consider that AIs can see now:

> Vamp bites
> Vamp glares
> Vamp Screeches (sort of apparent with windows, but still)
> Extendy proboscis fun times (Apparent when a corpse goes grey, but clothed is a good way to hide that)
> Lings popping back to life

**Comment by JustSumBody** [comment]
Better throw up them controversial and discussion labels

**Comment by Exxion** [comment]
> Implying anyone does any of those things in view of cameras anyway

**Comment by Ephx1** [comment]
They do quite often. People don't seem to know how far reaching some cameras are or that some cameras even exist. Vampires take lots of comfort in knowing that an AI can't actually see what they're doing. Two guys facing each other isn't suspicious, but two guys facing each other and one starts having glowy eyes, then shit's going down. Same applies to vampire succ, kidnapping someone and having a bite is completely lost on an AI and for all intents and purposes looks like a simple kidnapping between humans.

**Comment by Exxion** [comment]
If I'm not mistaken, cameras have the same view range as people normally do
Or at least they're supposed to

**Comment by AutismSpider** [comment]
only X Ray cams have  more view range per-say (only seen mommis make X Ray cams any way)

lets the Ai/any one using that cam to see past walls etc

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 1]
```diff
@@ -191,7 +191,7 @@ rcd light flash thingy on matter drain
 
 	ai_flags |= HIGHRESCAMS
 
-	eyeobj.addHear()
+	eyeobj.hires = 1
```
Comment (clusterfack) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: hires?

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 2]
```diff
@@ -111,7 +111,7 @@ var/list/ai_list = list()
 
 	aiMulti = new(src)
 	aicamera = new/obj/item/device/camera/ai_camera(src)
-
+	eyeobj.addHear()
```
Comment (clusterfack) [inline code/modules/mob/living/silicon/ai/ai.dm]: Should be done by a flag

Comment (JustSumBody) [inline code/modules/mob/living/silicon/ai/ai.dm]: How do you mean, just eyeobj.flags = HEAR or something?
There's this thing in addHear
`getFromPool(/mob/virtualhearer, src)`

I don't really understand, would I still need to do that somewhere as well?

Comment (clusterfack) [inline code/modules/mob/living/silicon/ai/ai.dm]: No, as long as New() calls ..() up to atom/movable/New(), having flags = HEAR is enough to make sure it gets a virtualhearer

File: code/modules/mob/living/silicon/ai/freelook/eye.dm [hunk 3]
```diff
@@ -42,7 +42,7 @@
 
 //An AI eyeobj mob cant hear unless it updates hires with a Malf Module
```
Comment (ComicIronic) [inline code/modules/mob/living/silicon/ai/freelook/eye.dm]: Update this comment.

File: code/modules/mob/mob.dm [hunk 4]
```diff
@@ -346,9 +346,16 @@ var/global/obj/screen/fuckstat/FUCK = new
 // message is output to anyone who can see, e.g. "The [src] does something!"
 // blind_message (optional) is what blind people will hear e.g. "You hear something!"
 /atom/proc/visible_message(var/message, var/blind_message, var/drugged_message, var/blind_drugged_message)
-	if(world.time>resethearers) sethearing()
+	if(world.time>resethearers)
+		sethearing()
 	for(var/mob/virtualhearer/hearer in viewers(get_turf(src)))
-		if(istype(hearer.attached, /mob))
+		if(istype(hearer.attached, /mob/camera/aiEye))
```
Comment (ComicIronic) [inline code/modules/mob/mob.dm:352]: These three typecheck blocks are a mess and should be handled by a proc on atoms.

**Comment by Ephx1** [comment]
hhhhhhh
Comments by people who are too lazy to properly comment here.
P.S. the bottom poster is dumb

**Comment by ghost** [comment]
But this is dumb, AI's will just valid hunt to the max now

**Comment by clusterfack** [comment]
@Ephx1 I don't know about 'nerfing' the module, but that is how the module works

**Comment by JustSumBody** [comment]
@clusterfack 
It isn't though, the high res cameras only let you read lips. Currently there is nothing that lets you see visible_messages as an AI unless they're within range of the physical core.
The malf module attaches a hearer to the AI eye, but visible_message doesn't take AI eyes into account. You end up with eyeobj.showmessage(visible message stuff) or whatever, but that doesn't go to anyone. You have to specify it to go to the eyeobj's assigned AI, which is what I've done here.

**Comment by clusterfack** [comment]
Seems I'm thinking of holopads when I added visible_message to them

**Comment by JustSumBody** [comment]
#11151
I can't explain

**Pull request closed without merging**